_shared/project/.github/workflows/ci.yml

```diff
@@ -110,6 +110,10 @@ jobs:
       - run: tox -e checkformatting
       {% else %}
       - run: tox -e {{ job }}
+      {% if job == 'tests' %}
+        env:
+          COVERAGE_FILE: .coverage.{% raw %}${{ matrix.python-version }}{% endraw +%}
+      {% endif %}
```

Comment:
For Python packages CI runs the tests in each version of Python that the package supports, each as a separate parallel job. This `COVERAGE_FILE` envvar makes each test job use a different coverage data file name so that all the files can later be combined by the coverage job. This is necessary for cases when a line of code is only used in a one version of Python, we need to record the coverage for each Python version and then combine them otherwise we'd get a coverage miss. This used to happen automatically when we used `coverage` directly but now that we use `pytest-cov` we need this envvar.

https://coverage.readthedocs.io/en/latest/cmd.html#data-file